### ISSUE TITLE
source update

### ISSUE THREAD

**Issue by mimtalendcloud** [StamusNetworks/scirius#174]
I have problem with scirius. When I want to add new source, error is shown:
![sc](https://user-images.githubusercontent.com/47530328/54692481-644aaf00-4b25-11e9-945a-8279760c758b.JPG)
During installation I followed steps in oficial documentation: https://scirius.readthedocs.io/en/latest/installation-ce.html

**Comment by pevma** [comment]
Did you just enable the public source ? (and the err appeared?)
I am also assuming there was no problem with internet connection? (just to double check ).

**Comment by mimtalendcloud** [comment]
Internet connection is stable. There is no problem with downloading rules using suricata-update. 
Should I set directory for rules? Or is it set by default?

**Comment by pevma** [comment]
From the screenshot it seems this is an older version (3.0.1) - the current should be 3.2.0
Could you try to update please and see if the issue persists?

Thank you 

> -- 
> Regards,
> Peter Manev

**Comment by mimtalendcloud** [comment]
![sc](https://user-images.githubusercontent.com/47530328/54935845-6f388180-4f21-11e9-860b-a4ab73694fa5.JPG)
I have updated version. The problem persist.
My tech specifications:
OS: Debian 9
Webpack: 3.11.0
Npm: 6.9.0
node-sass: 4.7.2
django: 1.11.20

**Comment by mimtalendcloud** [comment]
(path/to)/scirius/rules/views.py:1256: YAMLLoadWarning: calling yaml.load() without Loader=... is deprecated, as the default Loader is unsafe. Please read https://msg.pyyaml.org/load for full details.
  public_sources = yaml.load(yaml_data)

maybe this may cause the problem? I have tried to read url, but no idea what to do with this warning :/

**Comment by pevma** [comment]
And you have Scirius.rules inside yaml specified correctly , right ?

> -- 
> Regards,
> Peter Manev

**Comment by mimtalendcloud** [comment]
do you think in suricata.yaml?

this is in my suricata.yaml file:
default-rule-path: /home/branislav/scirius/rules/
rule-files:
 - scirius.rules

**Comment by mimtalendcloud** [comment]
Can not delete directory
[26/Mar/2019 08:33:26] "POST /rules/source/2/update HTTP/1.1" 200 68

this message is from console after trying to update rules 

![sc](https://user-images.githubusercontent.com/47530328/54982222-9b4c1500-4faa-11e9-9803-36904813cb30.JPG)

and this is from web interface

**Comment by regit** [comment]
Suricata is required for the test. If it is installed in a custom place there is a variable that can be used in local_settings.py.

**Comment by mimtalendcloud** [comment]
I have used default place for suricata.
path to suricata.yaml file is /etc/suricata/suricata.yaml

**Comment by pevma** [comment]
@mimtalendcloud  - do you have Suricata installed ?

**Comment by mimtalendcloud** [comment]
yes I do. Suricata is working correctly.

**Comment by pevma** [comment]
Where is it installed ? `/usr/bin/suricata ` ?

**Comment by mimtalendcloud** [comment]
> 
> 
> Where is it installed ? `/usr/bin/suricata ` ?

yes

**Comment by mimtalendcloud** [comment]
![sc](https://user-images.githubusercontent.com/47530328/54984698-b40af980-4faf-11e9-9b95-409bc6991840.JPG)

**Comment by pevma** [comment]
Does the problem still persist or  you managed to solve it ? (it is not permissions or something similar correct?)

**Comment by mathieurbl** [comment]
I have the same problem.
Moreover I tried to add a custom rule and it's not working anymore.
For that I have in my rules directory the file "scirius.rules" with the following rule : 
![image](https://user-images.githubusercontent.com/47533125/56139361-029d2980-5f99-11e9-9a31-a13a4e794568.png)
When I'm trying to import this rule I fill all the fields like that :
![image](https://user-images.githubusercontent.com/47533125/56138977-3b88ce80-5f98-11e9-8ca5-adef7aba4c23.png)
But when I hit Submit the upload field is empty and I get this : 
![image](https://user-images.githubusercontent.com/47533125/56139152-9a4e4800-5f98-11e9-8c70-a6e3a84a31e9.png)

**Comment by regit** [comment]
Can you activate DEBUG mode by setting `DEBUG=True` in local_settings.py ? And then post the traceback.

**Comment by mathieurbl** [comment]
I already activated DEBUG mode but I don't know where I can find the traceback.

**Comment by pevma** [comment]
Hi,

On the web interface - does it display something more ?

> -- 
> Regards,
> Peter Manev

**Comment by mathieurbl** [comment]
Hi, 
It's doesn't display something more and the problem persist : 
![image](https://user-images.githubusercontent.com/47533125/56187799-117cee00-6024-11e9-9f4c-abcf2823ebfa.png)

**Comment by regit** [comment]
OK so there is no traceback. Can the scirius write to /tmp/ ?

**Comment by mathieurbl** [comment]
No, there is no file or directory linked to scirius in my /tmp/

**Comment by pevma** [comment]
Does Scirius have permissions to write in that folder though ?

**Comment by Rajikoro** [comment]
I am experiencing exactly the same problem.
There are rights to `/tmp`.

**Comment by Rajikoro** [comment]
This doesn't seem like a rights issue, maybe I did `chmod -R 777 /` ))))))

**Comment by Rajikoro** [comment]
Works on the latest version from the repository, but there are enough other problems
